packages/editor/src/components/rich-text/index.js

```diff
@@ -229,7 +229,11 @@ export class RichText extends Component {
 	createRecord() {
 		const range = getSelection().getRangeAt( 0 );
 
-		return create( {
+		if ( range === this.createRecord.lastRange ) {
```

Comment:
Isn't it possible that the range stays the same but the value changes? (say using a custom control )

Reply:
What would that custom control do? We're getting the range straight from the browser. According to the API documentation, the reference should be the same if the selection didn't change:

> Thus subsequent calls of this method returns the same range object if nothing has removed the context object's range in the meantime. In particular, getSelection().getRangeAt(0) === getSelection().getRangeAt(0) evaluates to true if the selection is not empty.
>
> https://w3c.github.io/selection-api/#dom-selection-getrangeat

Reply:
Let's say a custom control inverting a word.

Reply:
I don't think it matters as the DOM updated (new text nodes). I can check.